src/robust.jl

```diff
@@ -0,0 +1,133 @@
+# Robust Statistics
+
+#############################
+#
+#   Trimming outliers
+#
+#############################
+
+# Trimmed set
+"""
+    trim(x, [prop=0.0, count=0])
+
+
+Remove either `count` or proportion `prop` of the highest and lowest
+elements in `x` and return the result. To compute the trimmed mean of
+`x`, use `mean(trim(x, prop=prop))`; to compute the variance of this
+quantity, use `trimvar(x, prop=prop)`.
+
+# Example
+```julia
+julia> trim([1,2,3,4,5], prop=0.2)
+3-element Array{Int64,1}:
+ 2
+ 3
+ 4
+```
+"""
+function trim(x::AbstractVector; prop::Real=0.0, count::Integer=0)
+    trim!(copy(x); prop=prop, count=count)
+end
+
+"""
+    trim!(x, [prop=0.0, count=0])
+
+A variant of `trim` that modifies `x` in place.
+"""
+function trim!(x::AbstractVector; prop::Real=0.0, count::Integer=0)
+    n = length(x)
+    n > 0 || throw(ArgumentError("x can not be empty."))
+
+    if count == 0
+        0 <= prop < 0.5 || throw(ArgumentError("prop must satisfy 0 ≤ prop < 0.5."))
+        count = floor(Int, n * prop)
+    else
+        0 <= count < n/2 || throw(ArgumentError("count must satisfy 0 ≤ count < length(x)/2."))
+    end
```

Comment:
the `if count == 0` code block appears three times, and each is identical.  might consider pulling it out into a separate function and using `@inline` so there is no overhead.